File: tests/fixtures/typescript/casts.ts
import { AuditLog, UserRepository } from "./repos";

export function first(found: unknown, id: number): void {
  const repo = found as UserRepository;
  void repo.deleteUser(id);
  const audit = found as unknown as AuditLog;
  audit.deleteUser(id + 1);
  void (found as UserRepository).deleteUser(id + 2);
  const loose = found as any;
  loose.deleteUser(id + 3);
  const partial = found as Partial<UserRepository>;
  void partial.deleteUser?.(id + 4);
}

File: tests/fixtures/python/casts.py
import typing
from typing import cast

from repos import AuditLog, UserRepository


def first(found: object, user_id: int) -> None:
    repo = cast(UserRepository, found)
    repo.delete_user(user_id)
    audit = typing.cast("AuditLog", found)
    audit.delete_user(user_id + 1)
    cast(UserRepository, found).delete_user(user_id + 2)
    missing = cast("Missing", found)
    missing.delete_user(user_id + 3)
    loose = cast(typing.Any, found)
    loose.delete_user(user_id + 4)
